### PR TITLE
476 - 20200602 type hints part3

### PR DESCRIPTION
**Pull request by hjmjohnson**
Related to  #476 .

### Description
Next set of changes related to clean typehinting.

### Status
Ready

### Types of changes
- [X] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/spatial/array.py [hunk 1]
```diff
@@ -667,7 +667,9 @@ def __init__(
     def randomize(self):
         self._do_transform = self.R.random_sample() < self.prob
         if hasattr(self.min_zoom, "__iter__"):
-            self._zoom = (self.R.uniform(l, h) for l, h in zip(self.min_zoom, self.max_zoom))
+            # TODO: Review the types here
```
Comment (hjmjohnson) [inline monai/transforms/spatial/array.py:670]: This suppression is to allow for clean static checking for type hints.  This case will need to be revisited in the future, but suppressing this difficult type hint case temporarily in order to efficiently address more beneficial issues first.

**Comment by hjmjohnson** [comment]
@wyli Ready to go.

**Comment by wyli** [comment]
> @wyli Ready to go.

it probably passes your tests locally with your specific environment, but currently I couldn't verify the correctness...

I think we need to fix and merge https://github.com/Project-MONAI/MONAI/pull/483 before merging the rest of the relevant PRs. so, could you please help with 483 when you have time?

**Comment by hjmjohnson** [comment]
#483 is going to require all of the items of #473 before it will pass.  That is why I have been working all day to break apart #473 into into smaller pull requests.  The individual pull requests will be a nightmare to make pass pytype individually.

We spent almost a week identifying and addressing what needs to be done to get pytype, mypy, and flake8 to pass initial type hinting checks.

**Comment by wyli** [comment]
> #483 is going to require all of the items of #473 before it will pass. That is why I have been working all day to break apart #473 into into smaller pull requests. The individual pull requests will be a nightmare to make pass pytype individually.
> 
> We spent almost a week identifying and addressing what needs to be done to get pytype, mypy, and flake8 to pass initial type hinting checks.

thanks, since this is a significant new feature, would be great to document your efforts in ticket https://github.com/Project-MONAI/MONAI/issues/476, so that we know what the plan is and where we are...

**Comment by hjmjohnson** [comment]
> thanks, since this is a significant new feature, would be great to document your efforts in ticket #476, so that we know what the plan is and where we are...

Done.

**Comment by hjmjohnson** [comment]
No progress too much work to maintain.

**Pull request closed without merging**